src/features/comments.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type State = {
+  comments: Comment[],
+  loaded: boolean,
+  hasError: boolean,
+  visible: boolean,
+  submitting: boolean,
+};
+
+const initialState: State = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+  visible: false,
+  submitting: false,
+};
+
+export const initComments = createAsyncThunk('comments/fetch',
+  (postId: number) => {
+    return getPostComments(postId);
+  });
+
+export const addComment = createAsyncThunk('createComment/post',
+  (data: Omit<Comment, 'id'>) => {
+    const newComment = createComment(data);
+
+    return newComment;
+  });
+
+export const removeComment = createAsyncThunk('removeComment/delete',
+  (commentId: number) => {
+    deleteComment(commentId);
+
+    return commentId;
+  });
+
+const commentsSlice = createSlice({
+  name: 'comments',
+  initialState,
+  reducers: {
+    setVisible: (state, actions: PayloadAction<boolean>) => {
+      state.visible = actions.payload;
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(initComments.pending, (state) => {
+      state.loaded = false;
+
+      return state;
+    });
+
+    builder.addCase(initComments.fulfilled,
+      (state, action: PayloadAction<Comment[]>) => {
+        state.comments = action.payload;
+        state.loaded = true;
+
+        return state;
```

Comment:
There is no need to return 'state' in Redux Toolkit's 'createSlice'. The changes are made directly to the state object, returning it is redundant.